gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/PalantirJavaFormatProviderPlugin.java

```diff
@@ -18,36 +18,87 @@
-            conf.setVisible(false);
-            conf.setCanBeConsumed(false);
-
-            conf.defaultDependencies(deps -> {
-                deps.add(rootProject
-                        .getDependencies()
-                        .create(ImmutableMap.of(
-                                "group", "com.palantir.javaformat",
-                                "name", "palantir-java-format",
-                                "version",
-                                        JavaFormatExtension.class.getPackage().getImplementationVersion())));
-            });
+        String implementationVersion = JavaFormatExtension.class.getPackage().getImplementationVersion();
+
+        Configuration configuration = rootProject.getConfigurations().create(CONFIGURATION_NAME);
+        configuration.setDescription("Internal configuration for resolving the palantir-java-format implementation");
+        configuration.setVisible(false);
+        configuration.setCanBeConsumed(false);
+        configuration.setCanBeResolved(true);
+        configuration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format",
+                            "version",
+                            implementationVersion)));
+        });
+
+        Configuration nativeConfiguration = rootProject.getConfigurations().create(NATIVE_CONFIGURATION_NAME);
+        nativeConfiguration.setDescription(
+                "Internal configuration for resolving the palantir-java-format implementation");
+        nativeConfiguration.setVisible(false);
+        nativeConfiguration.setCanBeConsumed(false);
+        nativeConfiguration.setCanBeResolved(true);
+        nativeConfiguration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format-native",
+                            "version",
+                            implementationVersion,
+                            "classifier",
+                            String.format("nativeImage-%s", getCurrentOs()),
+                            "ext",
+                            "exe")));
+        });
+        nativeConfiguration
+                .getAttributes()
+                .attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, "executable-nativeImage");
+        rootProject.getDependencies().registerTransform(ExecutableTransform.class, transformSpec -> {
+            transformSpec.getFrom().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, "exe");
```

Comment:
How is this working? Does Gradle set this attribute to be the extension of the dependency?

Reply:
Yes, from my testing - I saw it mentioned in a few other places in their codebase: https://github.com/gradle/gradle/blob/2d2ed60fc306a1101536500489c243f2d240af68/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/attributes/immutable/artifact/ImmutableArtifactTypeRegistry.java#L118